daemon/src/taker_cfd.rs

```diff
@@ -164,108 +135,50 @@ impl<O, W> Actor<O, W> {
 }
 
 #[xtra_productivity]
-impl<O, W> Actor<O, W>
-where
-    O: xtra::Handler<
-            oracle::GetAnnouncement,
-            Return = Result<olivia::Announcement, oracle::NoAnnouncement>,
-        > + xtra::Handler<oracle::MonitorAttestation>,
-    W: xtra::Handler<wallet::BuildPartyParams, Return = Result<PartyParams>>
-        + xtra::Handler<wallet::Sign, Return = Result<PartiallySignedTransaction>>,
-{
-    async fn handle_take_offer(
-        &mut self,
-        msg: TakeOffer,
-        ctx: &mut xtra::Context<Self>,
-    ) -> Result<()> {
-        let TakeOffer {
-            order_id,
+impl Actor {
+    async fn handle(&mut self, msg: PlaceOrder) -> Result<OrderId> {
+        let PlaceOrder {
+            offer_id,
             quantity,
             leverage,
         } = msg;
 
-        let disconnected = self
-            .setup_actors
-            .get_disconnected(order_id)
-            .with_context(|| {
-                format!("Contract setup for order {order_id} is already in progress")
-            })?;
-
-        let (order_to_take, maker_offers) = self
+        let offer = self
             .current_maker_offers
             .clone()
             .context("No maker offers available to take")?
-            .take_order(order_id);
-
-        let order_to_take = order_to_take.context("Order to take could not be found in current maker offers, you might have an outdated offer")?;
-
-        // The offer we are instructed to take is removed from the
-        // set of available offers immediately so that we don't attempt
-        // to take it more than once
-        {
-            self.current_maker_offers.replace(maker_offers);
-            self.projection_actor
-                .send(projection::Update(self.current_maker_offers.clone()))
-                .await?;
-        }
-
-        if !order_to_take.is_safe_to_take(OffsetDateTime::now_utc()) {
-            bail!("The maker's offer appears to be outdated, refusing to take offer",);
+            .take_order(offer_id)
+            .0
+            .context("Offer to take could not be found in current maker offers, you might have an outdated offer")?;
+
+        // TODO: Consider removing this. Why can't we just place the
+        // order we want, if it's the want we selected in the UI? The
+        // maker should simply reject our request to place an order if
+        // it's outdated
```

Comment:
👍 I'm for removing this. maker should deal with outdated orders by just rejecting them. at least we have visibility how often it happens, otherwise we don't know that takers run into this frustrating problem.

Reply:
I disagree. The taker UI does not tell you that an offer is outdated, and I don't want to introduce this as part of this PR. If we want to remove this feature we can, but it should not be part if this changeset because it has notthing to do with it.
We introduces this to prevent the taker from taking outdated offers created by the maker (!) after we introduces replicating the offer in the daemon. This can still happen and this is a safety net for the taker.